airflow/models/dag.py

```diff
@@ -2312,6 +2327,19 @@ def set_edge_info(self, upstream_task_id: str, downstream_task_id: str, info: Ed
         """
         self.edge_info.setdefault(upstream_task_id, {})[downstream_task_id] = info
 
+    def validate_schedule_and_params(self):
+        for k, v in self.params.items():
+            if isinstance(v, Param):
+                if (
+                    v.default is None
+                    and self.schedule_interval is not None
+                    and ('type' not in v.schema or "null" not in v.schema['type'])
+                ):
```

Comment:
`"null" not in v.schema['type']` looks wrong -- shouldn't it be

```suggestion
                if (
                    v.default is None
                    and self.schedule_interval is not None
                    and ('type' not in v.schema or v.schema['type'] != "null")
                ):
```

Reply:
And by moving the schedule_interval check up we can simplify this to one line
```suggestion
                if v.default is None and ('type' not in v.schema or v.schema['type'] != 'null'):
```

Reply:
`type` can be an array, hence we are checking `in`